File: connect4/evaluators/baseline.py
"""Baseline evaluator using generic three-in-a-row scoring."""

from ..board import GameState
from ..heuristics import (
    center_control_score,
    terminal_state_bonus,
    three_in_row_score,
    window_pattern_score,
)


class BaselineThreesEvaluator:
    """Scores open threes without checking whether they are playable now."""

    def evaluate(self, state: GameState, player: int) -> float:
        score = 0.0
        score += terminal_state_bonus(state, player)
        score += 3.0 * center_control_score(state, player)
        score += 2.0 * window_pattern_score(state, player)
        score += 1.5 * three_in_row_score(state, player)
        return score

    def __repr__(self) -> str:
        return "Baseline"


EVALUATOR = BaselineThreesEvaluator()

File: evaluators/baseline.c
/*
 * Baseline evaluator: terminal awareness plus stronger positional control.
 */

#include "../heuristics.h"

float baseline_evaluate(const GameState *state, int player) {
    float score = 0.0f;

    score += terminal_state_bonus(state, player);
    score += 2.5f * center_control(state, player);

    return score;
}
